### PR TITLE
Enable sonarcloud

### PR DESCRIPTION
**Pull request by madurangasiriwardena**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
```
Comment (omindu) [inline .github/workflows/build.yml:17]: let's use v2.  https://github.com/actions/setup-java

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Maven packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
```
Comment (omindu) [inline .github/workflows/build.yml:31]:
```suggestion
          restore-keys: ${{ runner.os }}-m2-
```

**Review by pulasthi7 — APPROVED** [review]

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Maven packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2
```
Comment (omindu) [inline .github/workflows/build.yml:29]:
```suggestion
          path: ~/.m2/repository
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        with:
+          distribution: 'adopt' # See 'Supported distributions' for available options
+          java-version: '11'
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+      - name: Cache Maven packages
+        uses: actions/cache@v1
```
Comment (omindu) [inline .github/workflows/build.yml:27]: there's cache v2 IINM

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2-
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: mvn -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar
```
Comment (omindu) [inline .github/workflows/build.yml:36]: will this be building the project with tests ?

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
+name: Build
```
Comment (omindu) [inline .github/workflows/build.yml:1]:
```suggestion
name: PR Builder
```

File: .github/workflows/build.yml (hunk 5 above)
Comment (madurangasiriwardena) [reply .github/workflows/build.yml:36]: Seems to be yes.

**Pull request merged**